### PR TITLE
fix(scheduler): springify quartz jobs

### PR DESCRIPTION
**Pull request by marchello2000**
By making quartz jobs play nicely in the spring land, they can be made into components and participate in DI
This allows us to remove the hack to "request" various beans from within jobs

### CODE REVIEW

**Comment by marchello2000** [comment]
@robzienert bumping this up :)

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy [hunk 1]
```diff
@@ -67,6 +68,18 @@ class SchedulerConfiguration {
     return factoryBean
   }
 
+  @Bean
+  /**
+   * Job factory used to create jobs as beans on behalf of Quartz
+   */
```
Comment (robzienert) [inline echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy]: Docs should go above method/class annotations.

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy [hunk 2]
```diff
@@ -67,6 +68,18 @@ class SchedulerConfiguration {
...
+  JobFactory jobFactory(ApplicationContext applicationContext)
+  {
```
Comment (robzienert) [inline echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy]: We also don't newline opening squigglies.

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy (hunk 1 above)
Comment (marchello2000) [reply echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy]: 👍

File: echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy (hunk 2 above)
Comment (marchello2000) [reply echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy]: dunno what happened there

**Comment by marchello2000** [comment]
@robzienert : fixed up, ready for mergy-merge

**Pull request merged**